yoda/llava_data_prep/README.md

```diff
@@ -0,0 +1,115 @@
+
+Once building the table templates is complete, proceed to running [create_synthetic_tables](./scripts/create_synthetic_tables.py) with flags listed in the [Scripts](#scripts) section.
+
+# Scripts
+
+## Iteration for table formatting/templating
+
+[generate_tables.py](./scripts/generate_tables.py):  This is a utility script that be used to iterate on properly formatting tables as needed.  In particular, it allows users to test if their columns definitions and modified tsv formatted table provide the expected output correctly.  This script is only intended for testing and each table template should be eventually placed in [table_templates.json](./table_templates/table_templates.json).
+
+## Generate synthetic tables
+
+[create_synthetic_tables.py](./scripts/create_synthetic_tables.py): This script is used to generate synthetic tables, along with logging information for QC.  
+
+Flags:
+ - --name: The name of the run.  A folder will be created as "logs" and the name specified with the .log extension.  The logging is set to DEBUG level.
+ - --num-its: The number of iterations/synthetic tables to generate.  At present, only online inference is used with no concurrent requests used for batch generation.  
+ - --split: Accepts choices of "train" or "val".  At present, this does not control much, other than the naming of the json file created with synthetic prompts data.  There is likely to be contamination until better splitting logic is implemented.
+
+
+
+# Future work
+
+Future work will potentially include:
+- increased control for table augmentations.
+- increased controls of [albumentations](https://albumentations.ai/) image augmentations.
+- increased control over the output dpi (a random choice of threee resolutions is currently hard coded).
+- batch queueing for improved performance, depending on the model batch size and handling.
+- improved train/val split logic.
+- more templates as examples to start.
+
+# Current testing
+
+- Create virtual environment.
+- pip install yoda/llava_data_prep/requirements.txt
+- Note any system dependancies or issues installing the requirements.
+- Run yoda/llava_data_prep/notebooks/edgar_ingestion.ipynb and ensure it completes.  Check some of the outputs.
+- Run yoda/llava_data_prep/scripts/generate_tables.py and ensure yoda/synthetic_data/tmp/images/test_table.jpg is created.
+- Run:
+```bash
+python yoda/llava_data_prep/scripts/create_synthetic_tables.py --name test --num-its 4
+```
+and ensure yoda/llava_data_prep/test is created with a subfolder of synthetic images and yoda/llava_data_prep/test/annotations_train.json exists.
```

Comment:
Might want to change the language here to - "After completion yoda/llava_data_prep/test is created ....etc"
Currently I read this as something the user needs to ensure vs the output of the previous command.